library/Fission/Web/Auth.hs

```diff
@@ -45,16 +46,38 @@ user = do
   cfg <- ask
   return <| BasicAuthCheck \auth -> runRIO cfg <| checkUser auth
 
-checkUser :: (HasLogFunc cfg, MonadSelda (RIO cfg)) => BasicAuthData -> RIO cfg (BasicAuthResult User)
-checkUser (BasicAuthData username password) = do
-  mayUser <- findOne <| select User.users `suchThat` User.byUsername (decodeUtf8Lenient username)
-  maybe (pure NoSuchUser) checkPassword mayUser
+
+
+-- TODO: Should we do a case-insensitive query for username instead?
+--
+checkUser
+  :: MonadDatabase
+  => BasicAuthData
+  -> RIO cfg (BasicAuthResult User)
+
+checkUser (BasicAuthData authUsername authPassword) = do
+  (\u -> do
+    where_ (byUsername u)
+    return u
+  )
+    |> from
+    |> selectFirst
+    |> andThen (
+      maybe
+        (pure NoSuchUser)
+        checkPassword
+    )
```

Comment:
💅 

```haskell
checkUser
  :: MonadDatabase
  => BasicAuthData
  -> RIO cfg (BasicAuthResult User)
checkUser (BasicAuthData authUsername authPassword) user = do
  result <- selectFirst <| from do
    where_ (byUsername user)
    return user

  maybe (pure NoSuchUser) checkpassword result
```

This reads more like English, more like SQL, 

I totally get the urge to turn everything into a pipeline, but I think that it actually hurts readability in this case. Don't be afraid to bind a variable!

---

Alternate, that's also quite readable IMO:

```haskell
checkUser (BasicAuthData authUsername authPassword) user = do
  result <- selectFirst <| from do
    where_ (byUsername user)
    return user

  case result of
    Nothing     -> pure NoSuchUser
    Just dbUser -> checkPasword dbUser
```

Reply:
Yeah you're right, I wrote it this way initially because I thought it'd be more like Ecto, in which the `where` would be in the pipeline instead of the `do` block. But that's not case, so will rewrite 👍